contracts/FlowIDTableStaking.cdc

```diff
@@ -1287,58 +1311,24 @@ pub contract FlowIDTableStaking {
         return self.account.copy<Bool>(from: /storage/stakingEnabled) ?? false
     }
 
-    /// Gets an array of the node IDs that are proposed and approved for the next epoch
+    /// Gets an array of the node IDs that have committed sufficient stake
+    /// and are approved for the next epoch
     pub fun getProposedNodeIDs(): [String] {
-        var proposedNodes: [String] = []
 
-        let approvedList = FlowIDTableStaking.getApprovedList()
-        let approvedNodeIDs: {String: Bool} = {}
-        for id in approvedList {
-            approvedNodeIDs[id] = true
-        }
+        let proposedNodeIDs = FlowIDTableStaking.getApprovedList()
+
+        for nodeID in proposedNodeIDs.keys {
 
-        for nodeID in FlowIDTableStaking.getNodeIDs() {
             let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
-            let approved = approvedNodeIDs[nodeID] ?? false
 
             // To be considered proposed, a node has to have tokens staked + committed equal or above the minimum
             // Access nodes have a minimum of 0, so they need to be strictly greater than zero to be considered proposed
-            if self.isGreaterThanMinimumForRole(numTokens: self.NodeInfo(nodeID: nodeRecord.id).totalCommittedWithoutDelegators(), role: nodeRecord.role)
-               && approved
-            {
-                proposedNodes.append(nodeID)
-            }
-        }
-        return proposedNodes
-    }
-
-    /// Gets an array of all the nodeIDs that are staked.
-    /// Only nodes that are participating in the current epoch
-    /// can be staked, so this is an array of all the active
-    /// node operators
-    pub fun getStakedNodeIDs(): [String] {
-        var stakedNodes: [String] = []
-
-        let currentList = self.account.copy<[String]>(from: /storage/idTableCurrentList)
-            ?? panic("Could not get current list")
-        let currentNodeIDs: {String: Bool} = {}
-        for id in currentList {
-            currentNodeIDs[id] = true
-        }
-
-        for nodeID in FlowIDTableStaking.getNodeIDs() {
-            let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
-            let current = currentNodeIDs[nodeID] ?? false
-
-            // To be considered staked, a node has to have tokens staked equal or above the minimum
-            // Access nodes have a minimum of 0, so they need to be strictly greater than zero to be considered staked
-            if self.isGreaterThanMinimumForRole(numTokens: nodeRecord.tokensStaked.balance, role: nodeRecord.role)
-               && current
+            if !self.isGreaterThanMinimumForRole(numTokens: self.NodeInfo(nodeID: nodeRecord.id).totalCommittedWithoutDelegators(), role: nodeRecord.role)
             {
-                stakedNodes.append(nodeID)
+                proposedNodeIDs[nodeID] = nil
             }
         }
-        return stakedNodes
+        return proposedNodeIDs.keys
```

Comment:
So, since this returns `keys` here this would still include nodes that don't meet staking requirements. Which kinda makes the above loop not super useful, no?

Reply:
if an entry in a dictionary is set to `nil`, then it won't be included in the `keys` list any more afaik. I should test that though

Reply:
Seems like Josh is right, I checked this in a [Playground](https://play.flow.com/local-project?type=script&id=local-script-temp-0&storage=none).

I think we could change the code to make the behaviour more obvious -> **Suggestion:** use `proposedNodeIDs.remove(key: nodeID)` instead of setting to nil

Reply:
I refactored this method in another PR anyway so we don't use this any more. I changed the other usage of it though to use `remove`